### DISCUSSION TITLE
A request of help about the package and article

### DISCUSSION THREAD

**Discussion by LynnChan99** [saezlab/mistyR#29] [Q&A]
Dear professor,

Sorry for bothering you at first.Recently,I was trying to apply misty into my spatial data analysis,however,some questions came to me and I found no mentioned information about it in your article or github,so I decided to write this letter to seek for your help.

Firstly,I don't understand the output file 'coefficients',only targets exist in this .txt file.Does it mean,for example,the mean intra importance of TGFB1.TGFBR2(A pair of ligand-receptor of my data) as target?and what's meaning of p value?It seems should be 0 of all of them in the intra view,but it's not like that.

Secondly,you mentioned the intercept means environmental effects on the mean expression of the targets in your article,it confused me why there exists environmental effects?It would be explicit if you could make an example about this notion please.

Lastly,I wanna to confirm whether my understanding about the predictor and target is right.In my view,the predictor means the input gene expression in intra/juxta/para view,and the targets mean the rest gene sets in the same view?

It would be my honor if I could get help from you.Thank you,Wish you all best!

**Comment by LynnChan99** [comment]
Besides,I noticed that mentioned two breast cancer 10✖️genomics sections were used in your analysis,but only one sections showed in your mistyr websites.My question is that you just analyzed these two sections spearately or tried to merge them into one dataset(if so,do I need to correct the batch effect or just merge them) in the final intra/juxta/para view?

**Comment by jtanevski** [comment]
All outputs are contained in the results folder defined when running the `run_misty` function. They consist of a file coefficients.txt containing per-target information about the coefficients of the meta-model, a file performance.txt containing per-target performance information about a model trained on the intraview only and a model trained using all views, and one importances_*.txt file per view and target, with raw importance metrics. 

If I understand your example correctly, you have a target named TGFB1.TGFBR2. You are attempting to model the (joint?) expression of this pair of genes. In the file coefficients (or alternatively in the list element named contributions) you can find the parameter corresponding to each view in the meta-model used to predict the values of this variable. The p-values for each view correspond to the p-value of a test against a null hypothesis of that coefficient having a value of zero. In other words, the significance of the contribution of that view to the overall prediction of the meta-model.

The meta-model also contains an intercept term. In the paper we give an intuition for that term, explaining it as the residual environmental effect per target. Technically, in a linear model, the intercept is proportional to the mean of the target if all predictors are zero.

Regarding the predictors and targets. Each model has a single target and uses all other variables as predictors Target = F(Predictors). MISTy builds view-specific models. These models predict the expression of a target by using the information from all other variables in that view.

I recommend collecting your results in a variable with the `collect_results` function. In that way, your results will be more structured, you can manipulate them much easier and you can simply use the variable to plot various aspects of your results. 

In the example, we use only one slide. In the paper, we used the two available slides. MISTy was run independently on each slide and the results were then aggregated as described in the Importance weighting and result aggregation subsection in the Methods section of the paper. There is no need to batch-correct or merge the samples.

I hope this answers your questions.
